File: CivitasJava/civitas/Sorpresa_PorCasaHotel.java
package civitas;

import java.util.ArrayList;

public class Sorpresa_PorCasaHotel extends Sorpresa
{
    private int valor;

    Sorpresa_PorCasaHotel(int valor, String texto)
    {
        super(texto);
        this.valor=valor;
    }

    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos)
    {
        if(super.jugadorCorrecto(actual, todos))
        {
            super.informe(actual, todos);
            todos.get(actual).modificarSaldo(valor*todos.get(actual).cantidadCasasHoteles());
        }
    }

    @Override
    public String toString()
    {
        return this.texto;
    }
}

File: CivitasRuby/Sorpresa_PorCasaHotel.rb
module Civitas
    class Sorpresa_PorCasaHotel < Sorpresa
        public_class_method :new
        def initialize(valor, texto)
            super(texto)
            @valor=valor
        end

        def aplicarAJugador(actual, todos)

            if jugadorCorrecto(actual,todos)
                informe(actual, todos)
                todos[actual].modificarSaldo(@valor*todos[actual].cantidadCasasHoteles)
            end

        end

        def toString

            return @texto

        end

    end
end